src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyLanguageCoverageTest.java

```diff
@@ -0,0 +1,152 @@
+package org.jenkinsci.plugins.scriptsecurity.sandbox.groovy;
+
+import org.jenkinsci.plugins.scriptsecurity.sandbox.Whitelist;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.GenericWhitelist;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.jvnet.hudson.test.Issue;
+
+public class GroovyLanguageCoverageTest {
+
+    @Rule
+    public ErrorCollector errors = new ErrorCollector();
+
+    private void assertEvaluate(Whitelist whitelist, Object expected, String script) {
+        SandboxInterceptorTest.assertEvaluate(whitelist, expected, script, errors);
+    }
+
+    @Ignore("This fails on m.\"fruit${bKey}\" returning null due to JENKINS-46327")
+    @Issue("JENKINS-46327")
+    @Test
+    public void quotedIdentifiers() throws Exception {
+        // See http://groovy-lang.org/syntax.html#_quoted_identifiers
+        assertEvaluate(new GenericWhitelist(),
+                true,
+                "def m = [fruitA: 'apple', fruitB: 'banana']\n" +
+                        "assert m.fruitA == 'apple'\n" +
+                        "assert m.'fruitA' == 'apple'\n" +
+                        "assert m.\"fruitA\" == 'apple'\n" +
+                        "assert m.'''fruitA''' == 'apple'\n" +
+                        "assert m.\"\"\"fruitA\"\"\" == 'apple'\n" +
+                        "assert m./fruitA/ == 'apple'\n" +
+                        "assert m.$/fruitA/$ == 'apple'\n" +
+                        "def bKey = 'B'\n" +
+                        "assert m.\"fruit${bKey}\" == 'banana'\n" +
+                        "assert m.\"\"\"fruit${bKey}\"\"\" == 'banana'" +
+                        "assert m./fruit${bKey}/ == 'banana'\n" +
+                        "assert m.$/fruit${bKey}/$ == 'banana'\n" +
+                        "return true\n"
+        );
+    }
+
+    @Test
+    public void gStringWithClosure() throws Exception {
+        // see http://groovy-lang.org/syntax.html#_special_case_of_interpolating_closure_expressions
+        assertEvaluate(new GenericWhitelist(),
+                true,
+                "def number = 1\n" +
+                        "def eagerGString = \"value == ${number}\"\n" +
+                        "def lazyGString = \"value == ${ -> number }\"\n" +
+                        "assert eagerGString == 'value == 1'\n" +
+                        "assert lazyGString.toString() == 'value == 1'\n" +
+                        "number = 2\n" +
+                        "assert eagerGString == 'value == 1'\n" +
+                        "assert lazyGString == 'value == 2'\n" +
+                        "return true\n"
+        );
+    }
+
+    @Test
+    public void arithmeticOperators() throws Exception {
```

Comment:
I'm ok with that - I'd rather some redundancy than end up missing some coverage because I misremembered what was covered elsewhere.